_posts/2017-08-18-comment-verifier-l-orthographe-de-vos-docs-depuis-travis-ci.md

```diff
@@ -0,0 +1,134 @@
+
+```
+personal_ws-1.1 fr 3
+php
+javascript
+android
+```
+
+A noter que l'entête de ce fichier (première ligne) est importante : les deux derniers arguments correspondent à la langue et au nombre de mots.
+
+Ensuite pour utiliser ce dictionnaire personnel, vous devez ajouter cet argument à la commande : `--personal=./.aspell.en.pws`
+
+Comment exécuter cet outil depuis Travis CI ?
+=============================================
+
+Pour lancer `aspell` depuis Travis CI, vous devez installer ce package apt dans le container Travis. Pour ce faire, ajoutez les lignes suivantes dans le fichier `.travis.yml` :
+
+```yml
+addons:
+  apt:
+    packages:
+      - aspell
+      - aspell-en
+      - aspell-fr # qui correspond à votre langue
+```
+
+Ensuite, dans ce même fichier de configuration, vous pouvez exécuter votre script personnalisé grâce à cette ligne :
+
+```yml
+script: your_script.sh
+```
+
+Dans ce script, vous pouvez utiliser la variable d'environnement `$TRAVIS_COMMIT_RANGE`, disponible dans le container Travis, pour récupérer seulement les fichiers qui ont changés dans la pull request du build en cours :
+
+```bash
+git diff --name-only $TRAVIS_COMMIT_RANGE
+```
+
+Si vous voulez récupérer seulement les fichiers de type markdown, vous pouvez ajouter `| grep .md` à la fin de la précédente ligne de commande.
+
+Une fois que vous avez les noms des fichiers que vous voulez vérifier sur votre pull request, vous pouvez lancer la commande `aspell list` que nous avons vue en première partie.
+
+Vous pouvez aussi utiliser les commandes `grep` et `sed` pour supprimer les meta-données ou les blocs de code de vos fichiers, avant d'exécuter la commande `aspell`, si vous ne voulez pas vérifier l'orthographe dans ces blocs.
+Par exemple, si vous voulez supprimer vos blocs de code de vos fichiers markdown, vous pouvez utiliser cette commande :
+
+```bash
+cat your_file.md | sed  -n '/^```/,/^```/ !p'
+```
+
+Comment envoyer les résultats vers votre pull request Github ?
+==============================================================
+
+Nous ne voulons pas que ce script bloque ceux qui souhaitent merger la pull request. La première chose à faire est donc de retourner `exit 0` à la fin de notre script qui sera exécuté depuis Travis CI. Sinon si un code d'erreur est retourné, Travis va indiquer un statut d'erreur sur la pull request qui empêchera de la merger.
+
+La façon la plus simple d'envoyer les résultats des précédentes commandes est donc de les poster dans un commentaire sur la pull request Github.
+
+Premièrement vous devrez choisir un utilisateur Github qui sera utilisé pour ajouter un commentaire, et configurer un token d'accès pour cet utilisateur :
+- Connectez vous sur [https://github.com](https://github.com) avec et utilisateur
+- Allez sur [https://github.com/settings/tokens](https://github.com/settings/tokens)
```

Comment:
I think you should give this link here: `https://github.com/settings/tokens/new` (because you don't ask to click on "Generate new token" button and it's a kind of shortcut)